code/controllers/subsystem/processing/quirks.dm

```diff
@@ -22,11 +23,68 @@ PROCESSING_SUBSYSTEM_DEF(quirks)
 		var/datum/quirk/T = V
 		quirks[initial(T.name)] = T
 		quirk_points[initial(T.name)] = initial(T.value)
+		quirk_names_by_path[T] = initial(T.name)
 
-/datum/controller/subsystem/processing/quirks/proc/AssignQuirks(mob/living/user, client/cli, spawn_effects)
+/datum/controller/subsystem/processing/quirks/proc/AssignQuirks(mob/living/user, client/cli, spawn_effects, roundstart = FALSE, datum/job/job, silent = FALSE, mob/to_chat_target)
 	GenerateQuirks(cli)
-	for(var/V in cli.prefs.character_quirks)
+	var/list/quirks = cli.prefs.character_quirks.Copy()
+	var/list/cut
+	if(job && job.blacklisted_quirks)
+		cut = filter_quirks(quirks, job)
+	for(var/V in quirks)
 		user.add_quirk(V, spawn_effects)
+	if(!silent)
+		to_chat(to_chat_target || user, "<span class='boldwarning'>The following quirks were cut during assignment either due to a job restriction or another reason: [english_list(cut)].</span>")
```

Comment:
This shows every single quirk the user has instead of just the ones that triggered the culling